### PR TITLE
Generate reference config

### PR DESCRIPTION
**Pull request by pglass**
## Changes proposed in this PR:
* Add a script to generate reference config from the JSON schema (`docgen` directory).
* Add descriptions to all the fields in the schema.json

## How I've tested this PR:

Ran the script, and reviewed the markdown.

## Checklist:
- [ ] Tests added
- [ ] CHANGELOG entry added

    [HashiCorp engineers only. Community PRs should not add a changelog entry.]::
    [Changelog entries should use present tense, e.g. "Add support for..."]::

### CODE REVIEW

File: docgen/main.go [hunk 1]
```diff
@@ -0,0 +1,153 @@
+// Generate markdown from the JSON schema for the Consul ECS config.
```
Comment (lkysow) [inline docgen/main.go:1]: in the k8s project i've been putting these things inside a `hack` folder. What do you think?

File: docgen/main.go [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+package main
```
Comment (lkysow) [inline docgen/main.go:2]: Can you put a comment at the top here (or a readme in this dir) that gives a brief description of what this script does and how to run it.

**Review by lkysow — APPROVED** [review]
Should we add a Makefile target too?

File: docgen/main.go (hunk 1 above)
Comment (pglass) [reply docgen/main.go:1]: Oh, fine with me. Maybe like `hack/generate-config-reference`?

Comment (lkysow) [reply docgen/main.go:1]: sounds good!

**Pull request merged**